### ISSUE TITLE
Cannot dismiss "Copilot is disabled" modal

### ISSUE THREAD

**Issue by kpitt** [zed-industries/zed#12740]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Once the "Copilot is disabled" modal is displayed on the screen, it cannot be dismissed without quitting Zed and restarting. Most editing functionality still works, but some functions and keyboard shortcuts seem to be blocked.

1. Disable Copilot in the `settings.json` file.

```json
  "features": {
    "inline_completion_provider": "none"
  },
  "show_copilot_suggestions": false,
```

2. Open the Welcome view (Help > Show Welcome).
3. Click on the "Sign in to GitHub Copilot" button. The "Copilot is disabled" modal will be displayed at the top of the window.
4. Hit `Esc` key, mouse click inside the modal, or mouse click outside the modal. The modal is still displayed.
5. Close the Welcome page, or just switch to a different editor tab. The modal is still displayed, and hides some of the content of the selected tab.
6. In the editor tab, inputting and editing text and many other editing actions work, but try for example `cmd + f` or `cmd + shift +f`, or select "Find" or "Find in Project" from the "Edit" menu. Nothing happens.

### Environment

Zed: v0.138.4 (Zed)
OS: macOS 14.5.0
Memory: 32 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

![Screenshot 2024-06-06 at 11 10 07 AM](https://github.com/zed-industries/zed/assets/16420/e02783c3-c8c0-4e4b-909d-9ff3302c2811)

**Comment by notpeter** [comment]
I do not believe this to be an issue in the current version Zed.  If you run into this please, file a clean issue with reproduction steps. Thanks!
